infor-design/enterprise-wc
# 1062 - IdsButton: change 'type' attribute to 'appearance'

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR changes IdsButton's API in a couple ways:
- The previous `type` attribute has been renamed to `appearance`.
- A new `type` attribute has been added that simply passes the value through to the HTMLButtonElement used as the shadow root container (for example, `type="submit"` now works on buttons).
- All files in this project referencing the type attribute (components/examples/docs/tests) have been updated to reflect these changes

**Related github/jira issue (required)**:
Closes #1062 

**Steps necessary to review your pull request (required)**:

_Test appearance changes:_
Smoke test all button-related components for accuracy:
- http://localhost:4300/ids-button/matrix.html
- http://localhost:4300/ids-button
- http://localhost:4300/ids-menu-button
- http://localhost:4300/ids-toggle-button (should appear "default")
- http://localhost:4300/ids-trigger-field/example.html (trigger buttons should appear "default")
- http://localhost:4300/ids-about/example.html
- http://localhost:4300/ids-message/example.html (two separately-styled buttons)
- http://localhost:4300/ids-action-sheet/example.html (see cancel button on mobile view)

_Test the new `type` setting:_
- Open http://localhost:4300/ids-form/example.html
- Use dev tools to inspect the Submit button.  The host element and the inner HTMLButtonElement should both have a `type` attribute of "submit":

![Screen Shot 2023-03-23 at 12 57 32 PM](https://user-images.githubusercontent.com/3249601/227278585-85540c43-e997-4f01-bae4-be5d6a19f446.png)

**Included in this Pull Request**:
